src/goto-instrument/accelerate/util.cpp

```diff
@@ -114,8 +116,10 @@ typet join_types(const typet &t1, const typet &t2)
     }
   }
 
-  std::cerr << "Tried to join types: "
-            << t1.pretty() << " and " << t2.pretty()
-            << '\n';
-  assert(!"Couldn't join types");
+#ifdef DEBUG
+  std::cout << "Failed to join types: "
+            << t1.pretty() << " and " << t2.pretty() << "\n";
+#endif
+
+  INVARIANT(false, "failed to join types");
```

Comment:
I do remain in favour of moving the entire error message into the `INVARIANT`. My rationale here is that those error messages are very frequent (I've got hundreds of them in Debian experiments). Of course  the code needs proper debugging, but doing so is much easier when sufficient context is provided.

Reply:
What about:
```
messaget().error() << "Failed to join types: " << t1.pretty() << " and " << t2.pretty() << "\n";
UNREACHABLE;
```

Reply:
Constructing a messaget without a message handler would cause the message to be simply dropped, I'm afraid.

Reply:
Then I'll just replace the original assert with an INVARIANT and leave the rest unchanged. That fixes the compilation error and does not make things worse.